kotlin-sqlobject/src/main/kotlin/org/jdbi/v3/sqlobject/kotlin/KotlinDefaultMethodHandlerFactory.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject.kotlin
+
+import org.jdbi.v3.core.kotlin.isKotlinClass
+import org.jdbi.v3.sqlobject.Handler
+import org.jdbi.v3.sqlobject.HandlerFactory
+import java.lang.reflect.Method
+import java.util.*
+
+class KotlinDefaultMethodHandlerFactory : HandlerFactory {
+
+    private val implsClassCache = Collections.synchronizedMap(WeakHashMap<Class<*>, Map<MethodKey, Method>>())
```

Comment:
`KotlinDefaultMethodHandlerFactory.buildHandler` will be invoked on every method of a Kotlin interface while building handler. The cache helps to avoid multiple searches through inner classes to determine if there is a default implementation.